rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1759,35 +1819,47 @@ public BooleanWrapper mapOnNodeUrlSet(Set<String> nodeUrls, Predicate<RMNode> op
         return new BooleanWrapper(result);
     }
 
-    private boolean internalUnlockNode(RMNode rmnode) {
-        if (!rmnode.isLocked()) {
-            logger.warn("Cannot unlock a node that is not locked: " + rmnode.getNodeURL());
+    boolean internalUnlockNode(RMNode rmNode) {
+        if (!rmNode.isLocked()) {
+            logger.warn("Cannot unlock a node that is not locked: " + rmNode.getNodeURL());
             return false;
         }
 
         try {
             // can throw a security exception if the caller is not an admin
-            this.checkNodeAdminPermission(rmnode, this.caller);
-            rmnode.unlock(this.caller);
+            this.checkNodeAdminPermission(rmNode, this.caller);
+            rmNode.unlock(this.caller);
 
             // an eligible node is a node that is free AND not locked
-            if (rmnode.isFree()) {
-                eligibleNodes.add(rmnode);
+            if (rmNode.isFree()) {
+                eligibleNodes.add(rmNode);
             }
+
+            updateNode(rmNode);
         } catch (Exception ex) {
             logger.warn("", ex);
```

Comment:
It's because I prefer to have a clear message instead of an empty line, even if the stack trace is put just below. You are right, I have updated the implementation to make the behaviour consistent.